core/src/prediction/InstrumentalPredictionStrategy.cpp

```diff
@@ -191,13 +189,16 @@ PredictionValues InstrumentalPredictionStrategy::precompute_prediction_values(
     if (std::abs(sum_weight) <= 1e-16) {
       continue;
     }
+    std::vector<double>& value = values[i];
+    value.resize(NUM_TYPES);
```

Comment:
This line is also moved to after
```
 if (std::abs(sum_weight) <= 1e-16) {
      continue;
    }
```
which is a bug in case you supply samples weights where some of them are zero (like would be the case where using sample weights for bandwith estimation), predictions would then be zero instead of empty